kadasrouting/gui/optimalroutebottombar.py

```diff
@@ -228,14 +228,19 @@ def calculate(self):
         allAreasToAvoidWGS = []
         if areasToAvoid:
             for areasToAvoidGeom in areasToAvoid:
-                areasToAvoidJson = json.loads(areasToAvoidGeom.asJson())
-                areasToAvoidWGS = []
-                for i, polygon in enumerate(areasToAvoidJson['coordinates']):
-                    areasToAvoidWGS.append([])
-                    for point in polygon:
-                        pointWGS = transformer.transform(point[0], point[1])
-                        areasToAvoidWGS[i].append([pointWGS.x(), pointWGS.y()])
-                allAreasToAvoidWGS.extend(areasToAvoidWGS)
+                try:
+                    areasToAvoidJson = json.loads(areasToAvoidGeom.asJson())
+                    areasToAvoidWGS = []
+                    for i, polygon in enumerate(areasToAvoidJson['coordinates']):
+                        areasToAvoidWGS.append([])
+                        for point in polygon:
+                            pointWGS = transformer.transform(point[0], point[1])
+                            areasToAvoidWGS[i].append([pointWGS.x(), pointWGS.y()])
+                    allAreasToAvoidWGS.extend(areasToAvoidWGS)
+                except AttributeError:
+                    # if the custom polygon button is checked, but no polygon has been drawn
+                    pushWarning(self.tr("Custom polygon button is checked, but no polygon is drawn"))
+                    return
```

Comment:
Probably it's better to just check if areasToAvoidGeom is None, and continue the loop. In this case, the user can get the route (without a polygon to be avoided).